src/main/java/me/moodcat/backend/rooms/RoomInstance.java

```diff
@@ -206,7 +217,8 @@ private void processVotes(final Song previousSong) {
     private void processNextSong(final Room room, final List<Song> history) {
         final List<Song> playQueue = room.getPlayQueue();
         if (playQueue.isEmpty()) {
-            playQueue.addAll(history);
+            VAVector vector = this.roomDAOProvider.get().findById(id).getVaVector();
+            playQueue.addAll(this.songDAOProvider.get().findForDistance(vector, NUMBER_OF_SELECTED_SONGS));
             history.clear();
```

Comment:
It doesnt make much sense to clear the history now, but we want to prevent it from growing to big as well. Cant we just limit its content size?